elastic/kibana
# [Fleet] Fix agent policy namespace validation

## Summary

Resolve https://github.com/elastic/kibana/issues/193507

Fix how we display validation errors, to show namespace validation errors even if the field is not touched. 

That PR fix also some response validation for the space prefix API as `managedBy` could be null.

## How to test?

With space awareness enabled (featureFlag + api call)

Create a policy with a custom namspace,

use the space API to add some restriction, verify the error message is now displayed in the agent policy editor

<img width="1512" alt="Screenshot 2024-11-14 at 10 36 47 AM" src="https://github.com/user-attachments/assets/78962101-81ba-43d7-af7a-30b6f29c7fc4">
